src/components/TradeAssetSearch/components/SearchTermAssetList.tsx

```diff
@@ -57,6 +65,14 @@ export const SearchTermAssetList = ({
     return assets.filter(asset => asset.chainId === activeChainId)
   }, [activeChainId, allowWalletUnsupportedAssets, assets, walletConnectedChainIds])
 
+  const handleAssetClick = useCallback(
+    (asset: Asset) => {
+      onAssetClick(asset)
+      fetchAndUpsertCustomTokenMarketData(asset.assetId)
```

Comment:
question: do we want to await market data before triggering `onAssetClick`? It would need loading state for decent UX but that would be available if `useFetchAndUpsertCustomTokenMarketData` was a react-query

Reply:
Removed this logic in https://github.com/shapeshift/web/pull/7370/commits/dedae61a38016ecb062e2d0350b53e20254d0f21 and instead will leverage the market data service when it's implemented.